vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -158,32 +216,11 @@ def pre_build(self):
                 )
             ]
 
-    def __call__(self, target_to_data_frame: dict[ModelID, pd.DataFrame]):
-        # Only relevant for a dynamic filter.
-        # Although targets are fixed at build time, the validation logic is repeated during runtime, so if a column
-        # is missing then it will raise an error. We could change this if we wanted.
-        # Call this from actions_utils
-        targeted_data = self._validate_targeted_data(
-            {target: data_frame for target, data_frame in target_to_data_frame.items() if target in self.targets},
-            eagerly_raise_column_not_found_error=True,
-        )
-
-        if (column_type := self._validate_column_type(targeted_data)) != self._column_type:
-            raise ValueError(
-                f"{self.column} has changed type from {self._column_type} to {column_type}. A filtered column cannot "
-                "change type while the dashboard is running."
-            )
-
-        # TODO: when implement dynamic, will need to do something with this e.g. pass to selector.__call__.
-        # if isinstance(self.selector, SELECTORS["numerical"] + SELECTORS["temporal"]):
-        #     options = self._get_options(targeted_data)
-        # else:
-        #     # Categorical selector.
-        #     _min, _max = self._get_min_max(targeted_data)
-
     @_log_call
     def build(self):
-        return self.selector.build()
+        # TODO: Align inner and outer ids to be handled in the same way as for other figure components.
+        selector_build_obj = self.selector.build()
+        return dcc.Loading(id=self.id, children=selector_build_obj) if self._dynamic else selector_build_obj
```

Comment:
Are there any UI improvements we should introduce until the "Universal Vizro placeholder component" comes?

Reply:
I think no, but I didn't actually try it out 😅 What are possible problems with how the UI looks now?

Reply:
Just as another data point, when testing especially (ie refreshing loads) it becomes quite apparent and odd to have these loading components even for the title

Reply:
Ah ok, I understand now. If the solution I suggested earlier in our chat doesn't work then yes, let's make a new ticket for this @petar-qb.

Reply:
Here's the solution I found -> https://github.com/mckinsey/vizro/commit/86171e0f4799a412c8c1d8818d237f3b938b2eea